app/cdap/components/NamespaceAdmin/SourceControlManagement/index.tsx

```diff
@@ -0,0 +1,200 @@
+  const [isUnlinkModalOpen, setIsUnlinkModalOpen] = useState(false);
+  const sourceControlManagement = useSelector((state) => state.sourceControlManagement);
+  const dispatch = useDispatch();
+  const toggleForm = () => {
+    setIsFormOpen(!isFormOpen);
+  };
+  const toggleUnlinkModal = () => {
+    setIsUnlinkModalOpen(!isUnlinkModalOpen);
+  };
+
+  const getBodyForSubmit = (formState, validate = false) => {
+    return {
+      test: validate,
+      repository: formState,
+    };
+  };
+
+  const getSourceControlManagement = () => {
+    MyNamespaceApi.getSourceControlManagement(params).subscribe({
+      next(res: ISourceControlManagement) {
+        // after getting the saved config, we need to fetch the token using the tokenName
+        MySecureKeyApi.getSecureData({ ...params, key: res.auth.tokenName }).subscribe((token) => {
+          res.auth.token = token;
+          dispatch({
+            type: NamespaceAdminActions.setSourceControlManagement,
+            payload: {
+              sourceControlManagement: res,
+            },
+          });
+        });
+      },
+      error(err) {
+        // may need to add more check whether the config doesn't exist or fetching config failed
+        dispatch({
+          type: NamespaceAdminActions.setSourceControlManagement,
+          payload: {
+            sourceControlManagement: null,
+          },
+        });
+      },
+    });
+  };
+
+  const onSubmit = (formState: ISourceControlManagement) => {
+    Observable.forkJoin(
+      MyNamespaceApi.setSourceControlManagement(params, getBodyForSubmit(formState)),
+      MySecureKeyApi.put(
+        { ...params, key: formState.auth.tokenName },
+        {
+          data: formState.auth.token,
+        }
+      )
+    ).subscribe({
+      next() {
+        getSourceControlManagement();
+      },
+      error(err) {
+        console.log('Error setting source control management: ', err);
+      },
+      complete() {
+        toggleForm();
+      },
+    });
+  };
+
+  const onDelete = () => {
+    Observable.forkJoin(
+      MyNamespaceApi.deleteSourceControlManagement(params),
+      MySecureKeyApi.delete({ ...params, key: sourceControlManagement.auth.tokenName })
+    ).subscribe({
+      next() {
+        getSourceControlManagement();
+      },
+      error(err) {
+        console.log('Error deleting source control management: ', err);
+      },
+      complete() {
+        toggleUnlinkModal();
+      },
+    });
+  };
+
+  useEffect(() => {
+    getSourceControlManagement();
+  }, []);
+
+  const actions = [
+    {
+      label: T.translate(`${PREFIX}.edit`),
+      actionFn: () => toggleForm(),
+    },
+    {
+      label: T.translate(`${PREFIX}.delete`),
+      actionFn: () => toggleUnlinkModal(),
+    },
+  ];
+
+  return (
+    <>
+      <StyledInfo>{T.translate(`${PREFIX}.info`)}</StyledInfo>
+      {!sourceControlManagement && (
+        <PrimaryContainedButton
+          children={T.translate(`${PREFIX}.linkButton`)}
```

Comment:
again?

Reply:
can you explain to my why you're doing children={T.translate} instead of doing > {T.translate} </PrimaryContainedButton> ??

Reply:
missed this comment, will address in subsequent pr